src/development/submodules.md

```diff
@@ -43,3 +43,33 @@ E: Error validating submodules in tree:
 ```
 
 Since the Platform.sh Git server cannot connect to Github via SSH without being granted an SSH key to do so, you should not be using an SSH URL: ``git@github.com:...``, but you should use an HTTPS URL instead: ``https://github.com/...``.
+
+## Use of private git repositories
+
+When using git submodules that are hosted on private repositories, using the `https` protocol will fail with errors like:
+
+```
+GitProtocolError: unexpected http resp 401 for https://bitbucket.org/myusername/mymodule.git/info/refs?service=git-upload-pack
+```
+
+To fix this, you need:
+
+1. To move your `.gitmodules` file from the https syntax to the ssh syntax, e.g.
+
+from:
+
+```
+[submodule "support/mymodule"]
+    path = support/mymodule
+    url = https://bitbucket.org/myusername/mymodule.git
+```
+
+to
+
+```
+[submodule "support/mymodule"]
+    path = support/mymodule
+    url=git@bitbucket.org:myusername/mymodule.git
+```
+
+2. to add a key in the project settings "Deploy Key" tab as per the [Private Repository](development/private-repository.md) documentation page, to allow our git service to pull the module from the remote git service.
```

Comment:
It's unclear from this sentence if users add a key to our settings or to the BitBucket settings page.

Reply:
to our settings

Reply:
Please update the text accordingly then so it's less unclear.